### PR TITLE
Fixed sentry errors

### PR DESCRIPTION
**Pull request by Gonzalo-Avalos-Ribas**
Users entity needed the `active` field, but because of the way we were calling the API, it was never provided. Also `firstname` and `lastname` are required strings, so I added a split. This is the same conversion code we have on the graph-azure project.
Group description must be a string because of constraints of the type. In the entity sometimes it returned null. I added a default `undefined` value.
Some of the logs show we get a 200 FetchError. In the logs i can see this is related to a timeout. This is really weird. But for now an easy fix should be to retry errors when response is 200.
Screenshot 2023-03-28 at 12 12 50

We were getting a key repetition on some devices, i added a guard condition to prevent this.
Lastly, the most prominent error happened when we would ingest an account that didn't have Intune active, the AD account step would fail, thats why I added a try catch statement.

### CODE REVIEW

File: src/steps/active-directory/group/converters.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export function createGroupEntity(data: Group): Entity {
         deletedOn: parseTimePropertyValue(data.deletedDateTime),
         classification: data.classification,
         createdOn: parseTimePropertyValue(data.createdDateTime),
-        description: data.description,
+        description: data.description ?? '',
```
Comment (zemberdotnet) [inline src/steps/active-directory/group/converters.ts:26]: I think converting `null` to an empty string is a bad pattern. In J1QL there are defined semantics for when a value is missing `value = undefined`, but searching for `value = ''` just seems wrong.

Comment (Gonzalo-Avalos-Ribas) [reply src/steps/active-directory/group/converters.ts:26]: Didn't know this! Thank you so much for the comment!

File: src/ms-graph/utils.ts [hunk 2]
```diff
@@ -8,7 +9,7 @@ export function isRetryable(err: any): boolean {
   // 504 GatewayTimeout
   // Others we could consider adding if necessary: 408, 503 and others
   const isRetryableStatusCode = (statusCode: number): boolean => {
-    return [-1, 400, 408, 500, 502, 503, 504].includes(statusCode);
+    return [-1, 200, 400, 408, 500, 502, 503, 504].includes(statusCode);
```
Comment (gastonyelmini) [inline src/ms-graph/utils.ts:12]: was curious on why 200 would be a status code to retry?

Comment (Gonzalo-Avalos-Ribas) [reply src/ms-graph/utils.ts:12]: I address this in the PR description: 
`Some of the logs show we get a 200 FetchError. In the logs i can see this is related to a timeout. This is really weird. But for now an easy fix should be to retry errors when response is 200.`

**Review by gastonyelmini — APPROVED** [review]
Looks good!

**Review by jroblesx — APPROVED** [review]
Look good @Gonzalo-Avalos-Ribas. I loved your comment specifying why the 200 code in the isRetryableStatusCode array.

**Review by jroblesx — APPROVED** [review]

**Pull request merged**